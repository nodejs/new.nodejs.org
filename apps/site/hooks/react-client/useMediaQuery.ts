'use client';

import { useState, useEffect } from 'react';

const useMediaQuery = (query: string) => {
  const [matches, setMatches] = useState(false);

  useEffect(() => {
    const { matches, addEventListener, removeEventListener } =
      window.matchMedia?.(query) ?? {
        matches: false,
        addEventListener: () => {},
        removeEventListener: () => {},
      };

    setMatches(matches);

    const handler = (event: MediaQueryListEvent) => setMatches(event.matches);

    addEventListener('change', handler);

    return () => removeEventListener('change', handler);
  }, [query]);

  return matches;
};

export default useMediaQuery;
