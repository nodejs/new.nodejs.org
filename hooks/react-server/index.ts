export { default as useCopyToClipboard } from './useCopyToClipboard';
export { default as useDetectOS } from './useDetectOS';
export { default as useIsCurrentPathname } from './useIsCurrentPathname';
export { default as useMediaQuery } from './useMediaQuery';
export { default as useNotification } from './useNotification';
export { default as useClientContext } from './useClientContext';
export { default as useBlogData } from './useBlogData';
export { default as useSiteNavigation } from './useSiteNavigation';
