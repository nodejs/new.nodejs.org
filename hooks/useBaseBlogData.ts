import { useMemo } from 'react';

import { blogData } from '@/next.json.mjs';

export const useBaseBlogData = (pathname: string) => {
  const { posts, pagination, categories } = blogData;

  const getPostsByCategory = (category: string) =>
    posts.filter(post => post.category === category);

  const getPostsByYear = (year: string) =>
    posts.filter(post => new Date(post.date).getFullYear() === Number(year));

  const getPagination = (currentYear: number | string) => {
    const _currentYear = Number(currentYear);

    return {
      next: pagination.includes(_currentYear + 1)
        ? _currentYear + 1
        : undefined,
      prev: pagination.includes(_currentYear - 1)
        ? _currentYear - 1
        : undefined,
    };
  };

  const currentCategory = useMemo(() => {
    // We split the pathname to retrieve the blog category from it
    // since the URL is usually /{languageCode}/blog/{category}
    // the third path piece is usually the category name
    const [, _pathname, category] = pathname.split('/');

    if (_pathname === 'blog' && category && category.length) {
      return category;
    }

    // if either the pathname does not match to a blog page
    // which should not happen (as this hook should only be used in blog pages)
    // or if there is no category in the URL we return the current year as category name
    // which is always the default category (for example, the blog index)
    return `year-${new Date().getFullYear()}`;
  }, [pathname]);

  return {
    posts,
    categories,
    currentCategory,
    getPostsByCategory,
    getPostsByYear,
    getPagination,
  };
};
