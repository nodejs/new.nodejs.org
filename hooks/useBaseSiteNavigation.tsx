import { useTranslations } from 'next-intl';
import type { RichTranslationValues } from 'next-intl';
import { useMemo } from 'react';

import { siteNavigation } from '@/next.json.mjs';
import type { NavigationEntry, NavigationKeys } from '@/types';

// These are mapped navigation entries. Navigation Entries can have sub-entries
type MappedItems = {
  text: ReturnType<ReturnType<typeof useTranslations>['rich']>;
  link: string;
  key: string;
  level: number;
  items: MappedItems[];
};

// Provides Context replacement for variables within the Link. This is also something that is not going
// to happen in the future with `nodejs/nodejs.dev` codebase
const replaceLinkWithContext = (link: string, context: RichTranslationValues) =>
  Object.entries(context).reduce(
    (finalLink, [find, replace]) =>
      finalLink.replace(
        `{${find}}`,
        typeof replace === 'string' ? replace : ''
      ),
    link
  );

const useBaseSiteNavigation = () => {
  const t = useTranslations();

  const mapNavigationEntries = (
    entries: Record<string, NavigationEntry>,
    context?: Record<string, RichTranslationValues>,
    level = 0
  ): MappedItems[] => {
    const getContext = (key: string) => (context && context[key]) || {};

    const getFormattedMessage = (translationId: string, key: string) =>
      t.rich(translationId, getContext(key));

    return Object.entries(entries).map(([key, item]) => ({
      text: getFormattedMessage(item.translationId, key),
      link: replaceLinkWithContext(item.link, getContext(key)),
      items: item.items
        ? mapNavigationEntries(item.items, context, level + 1)
        : [],
      level,
      key: key,
    }));
  };

  const rootNavigationItems = useMemo(
    () =>
      Object.entries(siteNavigation).reduce(
        (acc, [key, { translationId, link }]) => ({
          ...acc,
          [key]: { translationId, link },
        }),
        {}
      ),
    []
  );

  return {
    navigationItems: mapNavigationEntries(rootNavigationItems),
    getSideNavigation: (
      section: NavigationKeys,
      context?: Record<string, RichTranslationValues>
    ) => {
      const { items, translationId, link } = siteNavigation[section];

      return mapNavigationEntries(
        { [section]: { translationId, link }, ...items },
        context
      );
    },
  };
};

export default useBaseSiteNavigation;
