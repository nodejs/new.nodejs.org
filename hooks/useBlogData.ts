import { useCallback, useContext, useMemo } from 'react';
import { useRouter } from './useRouter';
import { BlogDataContext } from '../providers/blogDataProvider';

export const useBlogData = () => {
  const { asPath } = useRouter();

  const { posts, pagination, categories } = useContext(BlogDataContext);

  const getPostsByCategory = useCallback(
    (category: string) => posts.filter(post => post.category === category),
    [posts]
  );

  const getPostsByYear = useCallback(
    (year: number) =>
      posts.filter(post => new Date(post.date).getFullYear() === year),
    [posts]
  );

  const getPagination = useCallback(
    (currentYear: number) => ({
      next: pagination.includes(currentYear + 1) ? currentYear + 1 : undefined,
      prev: pagination.includes(currentYear - 1) ? currentYear - 1 : undefined,
    }),
    [pagination]
  );

  const currentCategory = useMemo(
    () => asPath.split('/')[3] || new Date().getFullYear().toString(),
    [asPath]
  );

  return {
    posts,
    categories,
    currentCategory,
    getPostsByCategory,
    getPostsByYear,
    getPagination,
  };
};
