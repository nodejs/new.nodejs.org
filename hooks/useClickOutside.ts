import { useEffect, useRef } from 'react';

type useClickOutsideType = (
  // eslint-disable-next-line no-unused-vars
  event: MouseEvent | TouchEvent
) => void;

export const useClickOutside = <T extends HTMLElement>(
  handler: useClickOutsideType
) => {
  const ref = useRef<T>(null);
  useEffect(() => {
    const clickEvent = 'ontouchstart' in window ? 'touchstart' : 'click';
    const listener = (event: MouseEvent | TouchEvent) => {
      if (!ref.current || ref.current.contains(event.target as Node)) {
        return;
      }
      handler(event);
    };

    document.addEventListener(clickEvent, listener, true);
    return () => document.removeEventListener(clickEvent, listener);
  }, [ref, handler]);
  return ref;
};
