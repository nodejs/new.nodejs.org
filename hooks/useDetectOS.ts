import { useEffect, useState } from 'react';
import { detectOS } from '../util/detectOS';
import { downloadUrlByOS } from '../util/downloadUrlByOS';

import type { UserOS } from '../types/userOS';

export const useDetectOS = () => {
  const [userOS, setUserOS] = useState<UserOS>('OTHER');
  const [bitness, setBitness] = useState('');

  useEffect(() => {
    setUserOS(detectOS());

    // This is necessary to detect Windows 11 on Edge.
    // [MDN](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/getHighEntropyValues)
    // [MSFT](https://learn.microsoft.com/en-us/microsoft-edge/web-platform/how-to-detect-win11)
    // @ts-expect-error no types for "userAgentData" because this API is experimental
    if (typeof navigator?.userAgentData?.getHighEntropyValues === 'function') {
      // @ts-expect-error no types for "userAgentData" because this API is experimental
      navigator.userAgentData
        .getHighEntropyValues(['bitness'])
        .then((ua: { bitness: string }) => setBitness(ua.bitness))
        .catch();
    }
  }, []);

  return {
    userOS,
    getDownloadLink: (version: string) =>
      downloadUrlByOS({
        userAgent: navigator?.userAgent,
        userOS,
        version,
        bitness,
      }),
  };
};
