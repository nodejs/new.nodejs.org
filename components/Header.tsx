'use client';

import Image from 'next/image';
import { useTranslations } from 'next-intl';
import { useTheme } from 'next-themes';
import type { FC, PropsWithChildren } from 'react';
import { useState } from 'react';

import Link from '@/components/Link';
import { useClientContext } from '@/hooks';
import { BASE_PATH } from '@/next.constants.mjs';
import { availableLocales } from '@/next.locales.mjs';

const Header: FC<PropsWithChildren> = ({ children }) => {
  const [showLangPicker, setShowLangPicker] = useState(false);
  const { resolvedTheme, setTheme } = useTheme();

  const { pathname } = useClientContext();
  const t = useTranslations();

  const toggleLanguage = t('components.header.buttons.toggleLanguage');
  const toggleTheme = t('components.header.buttons.toggleTheme');

  return (
    <header aria-label="Primary">
      <div className="container">
        <Link href="/" className="logo">
          <Image
            priority
            width="111"
            height="33"
            src={`${BASE_PATH}/static/images/logo.svg`}
            alt="Node.js"
          />
        </Link>

        {children}

        <div className="switchers">
          <button
            className="theme-switcher"
            type="button"
            title={toggleTheme}
            aria-label={toggleTheme}
            onClick={() =>
              setTheme(resolvedTheme === 'dark' ? 'light' : 'dark')
            }
          >
            <Image
              priority
              width="28"
              height="28"
              className="dark-image"
              src={`${BASE_PATH}/static/images/light-mode.svg`}
              alt="Theme Switcher"
            />

            <Image
              priority
              width="28"
              height="28"
              className="light-image"
              src={`${BASE_PATH}/static/images/dark-mode.svg`}
              alt="Theme Switcher"
            />
          </button>

          <button
            className="lang-picker-toggler"
            type="button"
            title={toggleLanguage}
            aria-label={toggleLanguage}
            onClick={() => setShowLangPicker(!showLangPicker)}
            aria-controls="lang-picker"
            aria-expanded="false"
          >
            <Image
              priority
              width="25"
              height="28"
              src={`${BASE_PATH}/static/images/language-picker.svg`}
              alt="Language Switcher"
            />
          </button>
        </div>

        {showLangPicker && (
          <ul className="lang-picker">
            {availableLocales.map(locale => (
              <li key={locale.code}>
                <Link
                  title={locale.name}
                  locale={locale.code}
                  href={pathname}
                  onClick={() => setShowLangPicker(false)}
                >
                  {locale.localName}
                </Link>
              </li>
            ))}
          </ul>
        )}
      </div>
    </header>
  );
};

export default Header;
