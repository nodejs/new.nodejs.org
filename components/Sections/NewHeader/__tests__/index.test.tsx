import userEvent from '@testing-library/user-event';
import { IntlProvider } from 'react-intl';
import { render, fireEvent, screen } from '@testing-library/react';
import Header from '..';
import { useMediaQuery } from '../../../../hooks/useMediaQuery';

let mockCurrentTheme = 'light';

const mockToggleTheme = jest.fn().mockImplementation(() => {
  mockCurrentTheme = mockCurrentTheme === 'dark' ? 'light' : 'dark';
});

// Mock dark mode module for controlling dark mode HOC behaviour
jest.mock('next-themes', () => ({
  useTheme: () => {
    return { theme: mockCurrentTheme, setTheme: mockToggleTheme };
  },
}));

// mock useRouter
jest.mock('next/router', () => ({
  useRouter() {
    return {
      locale: 'en',
    };
  },
}));

jest.mock('../../../../hooks/useLocale', () => ({
  useLocale: () => ({
    currentLocale: { code: 'en', name: 'English', localName: 'English' },
    availableLocales: [
      { code: 'en', name: 'English', localName: 'English' },
      { code: 'es', name: 'Spanish', localName: 'EspaÃ±ol' },
    ],
  }),
}));

// mock useMediaQuery hook for emulating mobile device
jest.mock('../../../../hooks/useMediaQuery', () => ({
  useMediaQuery: jest.fn(),
}));

describe('Tests for Header component', () => {
  beforeEach(() => {
    // @ts-ignore
    useMediaQuery.mockReturnValue(false);
  });

  afterEach(() => {
    mockToggleTheme.mockClear();
    mockCurrentTheme = 'dark';
  });

  it('renders shorter menu items for mobile', () => {
    mockCurrentTheme = 'dark';
    // @ts-ignore
    useMediaQuery.mockReturnValue(true);
    const { container } = render(
      <IntlProvider locale="en" onError={() => {}}>
        <Header />
      </IntlProvider>
    );

    expect(container).toMatchSnapshot();

    // @ts-ignore
    useMediaQuery.mockClear();
  });

  describe('Theme color switcher', () => {
    it('switches logo between light & dark', async () => {
      mockCurrentTheme = 'light';
      render(
        <IntlProvider locale="en" onError={() => {}}>
          <Header />
        </IntlProvider>
      );

      const lightLogo = screen.getByAltText('light-logo');
      expect(lightLogo).toBeInTheDocument();

      const toggle = screen.getByLabelText(
        'components.header.buttons.toggleDarkMode'
      );
      await userEvent.click(toggle);

      const darkLogo = screen.getByAltText('dark-logo');
      expect(darkLogo).toBeInTheDocument();
    });

    it('ignore key presses on color switcher', async () => {
      render(
        <IntlProvider locale="en" onError={() => {}}>
          <Header />
        </IntlProvider>
      );
      const toggler = screen.getByLabelText(
        'components.header.buttons.toggleDarkMode'
      );

      fireEvent.keyPress(toggler, { key: 'Enter', code: 13, charCode: 13 });

      expect(mockToggleTheme).toHaveBeenCalledTimes(0);
    });

    it('skips rendering in case no theme value was provided from plugin', () => {
      const { container } = render(
        <IntlProvider locale="en" onError={() => {}}>
          <Header />
        </IntlProvider>
      );
      expect(container).toMatchSnapshot();
    });
  });
});
