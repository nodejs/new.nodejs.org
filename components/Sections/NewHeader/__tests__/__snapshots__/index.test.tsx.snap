// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Header component Theme color switcher skips rendering in case no theme value was provided from plugin 1`] = `
<div>
  <nav
    aria-label="Primary"
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="startWrapper"
      >
        <a
          aria-label="Homepage"
          href="/"
        >
          <div
            class="logo"
          >
            <div>
              <img
                alt="light-logo"
                class="light-mode-only"
                data-nimg="1"
                decoding="async"
                height="33"
                loading="lazy"
                src="static/images/logo-light.svg"
                style="color: transparent;"
                width="111"
              />
            </div>
            <div>
              <img
                alt="dark-logo"
                class="dark-mode-only"
                data-nimg="1"
                decoding="async"
                height="33"
                loading="lazy"
                src="static/images/logo.svg"
                style="color: transparent;"
                width="111"
              />
            </div>
          </div>
        </a>
      </div>
      <ul
        class="tabs"
      >
        <li>
          <a
            href="/en/docs"
          >
            components.header.links.docs
          </a>
        </li>
        <li>
          <a
            href="/en/learn"
          >
            components.header.links.learn
          </a>
        </li>
        <li>
          <a
            href="/en/download"
          >
            components.header.links.download
          </a>
        </li>
      </ul>
      <div
        class="endWrapper"
      >
        <ul
          class="rightContainer"
        >
          <li
            class="searchBar"
          >
            <span
              class="sr-only"
            >
              Search Bar
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span>
              Search
            </span>
          </li>
          <li>
            <button
              aria-label="components.header.buttons.toggleDarkMode"
              aria-pressed="true"
              class="darkModeToggle"
              type="button"
            >
              <svg
                class="light-mode-only"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M14 2c1.82 0 3.53.5 5 1.35-2.99 1.73-5 4.95-5 8.65s2.01 6.92 5 8.65A9.973 9.973 0 0114 22C8.48 22 4 17.52 4 12S8.48 2 14 2z"
                />
              </svg>
              <svg
                class="dark-mode-only"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0a.996.996 0 000-1.41l-1.06-1.06zm1.06-10.96a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Tests for Header component renders correctly 1`] = `
<div>
  <nav
    aria-label="Primary"
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="startWrapper"
      >
        <a
          aria-label="Homepage"
          href="/"
        >
          <div
            class="logo"
          >
            <div>
              <img
                alt="light-logo"
                class="light-mode-only"
                data-nimg="1"
                decoding="async"
                height="33"
                loading="lazy"
                src="static/images/logo-light.svg"
                style="color: transparent;"
                width="111"
              />
            </div>
            <div>
              <img
                alt="dark-logo"
                class="dark-mode-only"
                data-nimg="1"
                decoding="async"
                height="33"
                loading="lazy"
                src="static/images/logo.svg"
                style="color: transparent;"
                width="111"
              />
            </div>
          </div>
        </a>
      </div>
      <ul
        class="tabs"
      >
        <li>
          <a
            href="/en/docs"
          >
            components.header.links.docs
          </a>
        </li>
        <li>
          <a
            href="/en/learn"
          >
            components.header.links.learn
          </a>
        </li>
        <li>
          <a
            href="/en/download"
          >
            components.header.links.download
          </a>
        </li>
      </ul>
      <div
        class="endWrapper"
      >
        <ul
          class="rightContainer"
        >
          <li
            class="searchBar"
          >
            <span
              class="sr-only"
            >
              Search Bar
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span>
              Search
            </span>
          </li>
          <li>
            <button
              aria-label="components.header.buttons.toggleDarkMode"
              aria-pressed="false"
              class="darkModeToggle"
              type="button"
            >
              <svg
                class="light-mode-only"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M14 2c1.82 0 3.53.5 5 1.35-2.99 1.73-5 4.95-5 8.65s2.01 6.92 5 8.65A9.973 9.973 0 0114 22C8.48 22 4 17.52 4 12S8.48 2 14 2z"
                />
              </svg>
              <svg
                class="dark-mode-only"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0a.996.996 0 000-1.41l-1.06-1.06zm1.06-10.96a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Tests for Header component renders shorter menu items for mobile 1`] = `
<div>
  <nav
    aria-label="Primary"
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="startWrapper"
      >
        <a
          aria-label="Homepage"
          href="/"
        >
          <div
            class="logo"
          >
            <div>
              <img
                alt="light-logo"
                class="light-mode-only"
                data-nimg="1"
                decoding="async"
                height="33"
                loading="lazy"
                src="static/images/logo-light.svg"
                style="color: transparent;"
                width="111"
              />
            </div>
            <div>
              <img
                alt="dark-logo"
                class="dark-mode-only"
                data-nimg="1"
                decoding="async"
                height="33"
                loading="lazy"
                src="static/images/logo.svg"
                style="color: transparent;"
                width="111"
              />
            </div>
          </div>
        </a>
      </div>
      <ul
        class="tabs"
      >
        <li>
          <a
            href="/en/docs"
          >
            components.header.links.docs
          </a>
        </li>
        <li>
          <a
            href="/en/learn"
          >
            components.header.links.learn
          </a>
        </li>
        <li>
          <a
            href="/en/download"
          >
            components.header.links.download
          </a>
        </li>
      </ul>
      <div
        class="endWrapper"
      >
        <ul
          class="rightContainer"
        >
          <li
            class="searchBar"
          >
            <span
              class="sr-only"
            >
              Search Bar
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span>
              Search
            </span>
          </li>
          <li>
            <button
              aria-label="components.header.buttons.toggleDarkMode"
              aria-pressed="true"
              class="darkModeToggle"
              type="button"
            >
              <svg
                class="light-mode-only"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M14 2c1.82 0 3.53.5 5 1.35-2.99 1.73-5 4.95-5 8.65s2.01 6.92 5 8.65A9.973 9.973 0 0114 22C8.48 22 4 17.52 4 12S8.48 2 14 2z"
                />
              </svg>
              <svg
                class="dark-mode-only"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0a.996.996 0 000-1.41l-1.06-1.06zm1.06-10.96a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
